cpp/src/lib/osd_access.cc

```diff
@@ -61,14 +61,38 @@ void OsdAccess::_remove_ctx(osd_t osd) {
   _osd_ctxs.erase(osd);
 }
 
-void OsdAccess::update(std::vector<std::pair<osd_t, info_caps>> &infos) {
-  ALBA_LOG(DEBUG, "OsdAccess::update");
-  std::lock_guard<std::mutex> lock(_osd_infos_mutex);
 
-  _osd_infos.clear();
-  for (auto &p : infos) {
-    _osd_infos.emplace(p.first, std::move(p.second));
-  }
+void OsdAccess::update(Proxy_client& client) {
+    if(!_filling){
+        ALBA_LOG(INFO, "OsdAccess::update:: filling up");
+        {
+            std::lock_guard<std::mutex> f_lock(_filling_mutex);
```

Comment:
don't you want/need to hold the lock for the entire duration of the `if(!_filling){` block?